### PR TITLE
Timeout configurable[Ready]

### PR DESCRIPTION
**Pull request by amrx101**
- `Timeout` is a `MqttOptions` variable
- A new member introduced to `EventLoop` struct of type `Duration`.
-  Default value is `5s`

### CODE REVIEW

File: rumqttc/src/eventloop.rs [hunk 1]
```diff
@@ -301,14 +306,14 @@ impl EventLoop {
...
         // mqtt connection with timeout
-        time::timeout(Duration::from_secs(5), async {
+        time::timeout(self.timeout, async {
             network.connect(connect).await?;
             Ok::<_, ConnectionError>(())
         })
         .await??;
 
         // wait for 'timeout' time to validate connack
-        let packet = time::timeout(Duration::from_secs(5), async {
```
Comment (amrx101) [inline rumqttc/src/eventloop.rs:311]: Should  `timeout` for `connect` and `ack` should be controlled by 2 different params?

Comment (tekjar) [reply rumqttc/src/eventloop.rs:311]: No need if we are returning the same error

**Comment by amrx101** [comment]
Tested with

- no TLS
- 1 way TLS
- 2 way TLS
- offline broker

@tekjar , this can be merged.

File: rumqttc/src/eventloop.rs [hunk 2]
```diff
@@ -265,7 +270,7 @@ impl EventLoop {
     }
 
     async fn network_connect(&mut self) -> Result<(), ConnectionError> {
-        let network = time::timeout(Duration::from_secs(5), async {
+        let network = time::timeout(self.timeout, async {
```
[the remark this answers is not in the document]
Comment (amrx101) [reply rumqttc/src/eventloop.rs:273]: @tekjar , could you elaborate a bit on this?

[the remark this answers is not in the document]
Comment (tekjar) [reply rumqttc/src/eventloop.rs:273]: I mean `mqtt_connect` which is calling this method already has a timeout (as far as I remember). We can probably remove this timeout. It's ok to let this be as it is as well and fix this later No problem either way

**Pull request merged**